### PR TITLE
Conformance2: add buffer overflow test for bindBufferRange

### PR DESCRIPTION
**Pull request by Richard-Yunchao**
buffer overflow will not lead to failure in bindBufferRange.

PTAL. Thanks.

### CODE REVIEW

File: sdk/tests/conformance2/buffers/buffer-overflow-test.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+debug("");
+var buffer1 = gl.createBuffer();
+gl.bindBufferRange(gl.UNIFORM_BUFFER, 0, buffer1, 0, 4); 
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, 
+    "Calling bindBufferRange to a buffer that no storage is allocated should success.");
+
+var buffer2 = gl.createBuffer();
+gl.bindBuffer(gl.UNIFORM_BUFFER, buffer2);
+gl.bufferData(gl.UNIFORM_BUFFER, 4, gl.STATIC_DRAW);
+gl.bindBufferRange(gl.UNIFORM_BUFFER, 0, buffer2, 0, 8);
```
Comment (zhenyao) [inline sdk/tests/conformance2/buffers/buffer-overflow-test.html]: UNIFORM_BUFFER has a situation where offset if not a multiple of UNIFORM_BUFFER_OFFSET_ALIGNMENT, you will get an error.  So here the test is not robust.

**Comment by Richard-Yunchao** [comment]
Yeah. The updated code use TRANSFORM_FEEDBACK_BUFFER instead of UNIFORM_BUFFER. PTAL, @zhenyao

File: sdk/tests/conformance2/buffers/buffer-overflow-test.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+var wtu = WebGLTestUtils;
+
+var gl = wtu.create3DContext(undefined, undefined, 2);
+
+debug("");
+var buffer1 = gl.createBuffer();
+gl.bindBufferRange(gl.TRANSFORM_FEEDBACK_BUFFER, 0, buffer1, 0, 4); 
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, 
+    "Calling bindBufferRange to a buffer that no storage is allocated should success.");
```
Comment (kenrussell) [inline sdk/tests/conformance2/buffers/buffer-overflow-test.html]: Typos/grammar:
to -> on
that -> where
success -> succeed

File: sdk/tests/conformance2/buffers/buffer-overflow-test.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+var buffer2 = gl.createBuffer();
+gl.bindBuffer(gl.TRANSFORM_FEEDBACK_BUFFER, buffer2);
+gl.bufferData(gl.TRANSFORM_FEEDBACK_BUFFER, 4, gl.STATIC_DRAW);
+gl.bindBufferRange(gl.TRANSFORM_FEEDBACK_BUFFER, 0, buffer2, 0, 8);
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, 
+    "Calling bindBufferRange to a buffer that the storage is not big enough should success.");
```
Comment (kenrussell) [inline sdk/tests/conformance2/buffers/buffer-overflow-test.html]: Typos/grammar:
to -> on
that -> where
success -> succeed

**Comment by kenrussell** [comment]
LGTM overall but Mo should re-review.

**Comment by zhenyao** [comment]
LGTM after fixing the nits kbr pointed out.

**Comment by Richard-Yunchao** [comment]
Oh... My poor English... @kenrussell and @zhenyao , Thanks for your review. 

The typos are addressed, PTAL.

**Comment by zhenyao** [comment]
LGTM

**Pull request merged**